source/Ship.cpp

```diff
@@ -3160,10 +3237,28 @@ int Ship::TakeDamage(vector<Visual> &visuals, const Weapon &weapon, double damag
 			ApplyForce((hitForce / distance) * d, weapon.IsGravitational());
 	}
 	
+	// Prevent various stats from reaching unallowable values.
+	hull = min(hull, attributes.Get("hull"));
+	shields = min(shields, attributes.Get("shields"));
+	// Weapons are allowed to overcharge a ship's energy or fuel, but code in Ship::DoGeneration()
+	// will clamp it to a maximum value at the beginning of the next frame.
+	energy = max(0., energy);
+	fuel = max(0., fuel);
+	heat = max(0., heat);
+	
 	// Recalculate the disabled ship check.
 	isDisabled = true;
 	isDisabled = IsDisabled();
 	
+	// Inflicted heat damage may also disable a ship, but does not trigger a "DISABLE" event.
```

Comment:
The DISABLE event is below on line 3254, so it looks like overheating now does trigger the DISABLE event.